js/web/lib/wasm/session-options.ts

```diff
@@ -64,6 +64,29 @@ const setExecutionProviders =
           case 'xnnpack':
             epName = 'XNNPACK';
             break;
+          case 'webnn':
+            epName = 'WEBNN';
+            if (typeof ep !== 'string') {
+              const webnnOptions = ep as InferenceSession.WebNNExecutionProviderOption;
+              if (webnnOptions?.deviceType) {
```

Comment:
[ignore if you change type of `deviceType` from `number` to `string`]

use `typeof webnnOptions?.deviceType !== 'undefined'` so that you won't miss when user set value to 0.